tests/test_return_value_policy_override.cpp

```diff
@@ -1,11 +1,75 @@
+#include <pybind11/smart_holder.h>
+
 #include "pybind11_tests.h"
 
 namespace test_return_value_policy_override {
 
 struct some_type {};
 
+struct obj {
```

Comment:
`obj` is usually used for instances, it's surprising as a type name.

The combinations of copyable/movable keep coming up, it would be good have some systematic naming that we could use throughout. What do you think about

```
struct type_cp1_mv1
struct type_cp1_mv0
struct type_cp0_mv1
struct type_cp0_mv0
```

?

With a comment like:
```
// cp = copyable, mv = movable, 1 = yes, 0 = no
```
Question that's maybe beyond this PR: do we have (or would it be safest) to add tests for the other two combinations?

IIUC here you have only `type_cp1_mv1` and `type_cp0_mv1`, is that correct?

Reply:
Thanks for your suggestions. I updated the PR with more tests.